packages/api/src/workers/check-video-streams.ts

```
@@ -0,0 +1,218 @@
+import axios, { AxiosPromise } from 'axios';
+import { Dictionary } from 'lodash';
+import { Connection, IsNull, Not } from 'typeorm';
+import { Reef } from '../reefs/reefs.entity';
+import { getYouTubeVideoId } from '../utils/urls';
+
+const logger = new Logger('CheckVideoStreams');
+
+type reefIdToYouTubeIdMap = Dictionary<{ reefId: number; url: string }>;
+
+// Create a basic structure for youTube response items
+interface YouTubeVideoItem {
+  id: string;
+  status: {
+    privacyStatus: 'private' | 'public' | 'unlisted';
+    embeddable: boolean;
+    uploadStatus: 'deleted' | 'failed' | 'processed' | 'rejected' | 'uploaded';
+  };
+  liveStreamingDetails: {
+    actualStartTime: string;
+    actualEndTime: string;
+  };
+}
+
+interface YouTubeApiResponse {
+  items: YouTubeVideoItem[];
+}
+
+interface SlackMessage {
+  channel: string;
+  blocks: {
+    type: string;
+    text?: {
+      type: string;
+      text: string;
+    };
+  }[];
+}
+
+const fetchVideoDetails = (
+  youTubeIds: string[],
+  apiKey: string,
+): AxiosPromise<YouTubeApiResponse> => {
+  return axios({
+    url: 'https://www.googleapis.com/youtube/v3/videos',
+    method: 'get',
+    params: {
+      key: apiKey,
+      id: youTubeIds.join(),
+      part: 'status,liveStreamingDetails',
+    },
+  });
+};
+
+const getError = (item: YouTubeVideoItem) => {
+  const { uploadStatus, privacyStatus, embeddable } = item.status;
+
+  if (privacyStatus === 'private') {
+    return 'Video is not public';
+  }
+
+  if (uploadStatus !== 'uploaded' && uploadStatus !== 'processed') {
+    return 'Video is no longer available';
+  }
+
+  if (!embeddable) {
+    return 'Video is not embeddable';
+  }
+
+  if (!item.liveStreamingDetails) {
+    return 'Video is not a live stream';
+  }
+
+  if (item.liveStreamingDetails.actualEndTime) {
+    return 'The live stream has ended';
+  }
+
+  if (!item.liveStreamingDetails.actualStartTime) {
+    return 'The live stream has not started yet';
+  }
+
+  return '';
```

returning undefined or null makes more sense here right? Then `getError() || ''`.

---

If we end up assigning an empty string in the returning variable, why return `null` in the first place? Also the function returns the error found. I can assume that an empty string means an empty error and not have to add extra logic.

---

My thinking is that if the function is called `getError` it should return an error, or *nothing* if there is none. We could be lenient here since JS treats `''` effectively as `undefined` since they're both falsey. 
However if we were in a stricter language (like Java), and were looking for the existence of an error in the first place, it would be more complex and less intuitive to check for an empty string than for null/undefined. A developer would have to read this function to know how to use it, and if they didn't read it at all it could introduce a bug.

I'm probably overthinking this haha, but it would be nice to either rename the function to `getErrorMessage` so it's more clear it only returns a string, or return `null/undefined` if there is no error. Let me know what you think :)

---

Ok I agree with changing to `getErrorMessage`.